src/benchmarklib/abstract_benchmark_item_runner.cpp

```diff
@@ -22,7 +23,12 @@ std::pair<std::vector<SQLPipelineMetrics>, bool> AbstractBenchmarkItemRunner::ex
     visualize_prefix = std::move(name);
   }
 
-  BenchmarkSQLExecutor sql_executor(_config->enable_jit, _sqlite_wrapper, visualize_prefix);
+  std::shared_ptr<const Table> expected_result_table = nullptr;
+  if (item_id < _dedicated_expected_results.size()) {
+    expected_result_table = _dedicated_expected_results[item_id];
+  }
+
+  BenchmarkSQLExecutor sql_executor(_config->enable_jit, _sqlite_wrapper, visualize_prefix, expected_result_table);
```

Comment:
The sql executor can be used for more than just a single SELECT (and will be in the TPC-C). This somewhat breaks its interface.

Reply:
I didn't get your point, yet. Could you explain the issue here more detailed? The _sqlite_wrapper is used in a similar way to the table newly integrated _expected_result_table: If the sqlite_wrapper is available, use the calculated expected output table and compare it with Hyrise's output table. Now, just don't do the calculation step via SQLite but use the provided expected result table for the comparison, if such a result table is provided.

Reply:
https://github.com/hyrise/hyrise/blob/78788d952236ada05857f8e0b6a97628a70d005f/src/benchmarklib/tpcc/procedures/tpcc_new_order.cpp#L56-L65

Here, the SQL executor is reused. As such, there is more than one result table. Providing a single expected result breaks that.

Reply:
Got it.